crates/oxc_linter/src/options/filter.rs

```diff
@@ -0,0 +1,272 @@
+use crate::RuleCategory;
+
+use super::{plugins::LintPlugins, AllowWarnDeny};
+use std::{borrow::Cow, fmt};
+
+/// Enables, disables, and sets the severity of lint rules.
+///
+/// Filters come in 3 forms:
+/// 1. Filter by rule name and/or plugin: `no-const-assign`, `eslint/no-const-assign`
+/// 2. Filter an entire category: `correctness`
+/// 3. Some unknow filter. This is a fallback used when parsing a filter string,
+///    and is interpreted uniquely by the linter.
+#[derive(Debug, Clone)]
+#[cfg_attr(test, derive(PartialEq))]
+pub struct LintFilter(AllowWarnDeny, LintFilterKind);
```

Comment:
I dislike tuple syntax, the next person need to convert it to `{}` syntax if a field is added.

This also creates a confusing `.0` `.1` below.